src/components/Facility/DischargedPatientsList.tsx

```diff
@@ -419,20 +438,18 @@ const DischargedPatientsList = ({
                     {
                       label: "Export Discharged patients",
                       action: async () => {
-                        const query = {
-                          ...params,
-                          csv: true,
-                        };
-                        const pathParams = { facility_external_id };
-                        const { data } = await request(
-                          routes.listFacilityDischargedPatients,
-                          {
-                            query,
-                            pathParams,
-                          },
-                        );
-                        return data ?? null;
+                        const result = await exportAsCSV();
+                        if (result.isError || !result.data) {
+                          return null;
+                        }
+                        const data = await result.data({
+                          signal: new AbortController().signal,
+                        });
+                        console.log(data);
+
+                        return data || null;
```

Comment:
what is this?

also you shouldn't be getting the `isError` fields and all. you should get the data directly. error handling is done by the global query client.

why call `data` and pass abort controller here?